### PR TITLE
Better message on missing underlying schema object

### PR DESCRIPTION
**Pull request by Sajan-99**
If the underlying schema has a missing object or is removed then this message will help clarify which field it is when looking at the error.

https://github.com/graphql-java/nadel/issues/157

### CODE REVIEW

**Comment by andimarek** [comment]
@Sajan-99 we should add a test for it. Have a look at NadelE2E test.

File: src/test/groovy/graphql/nadel/NadelE2ETest.groovy [hunk 1]
```diff
@@ -984,4 +986,66 @@ class NadelE2ETest extends Specification {
...
+                .query(query)
+                .operationName("OpName")
+                .build()
+
+        when:
+        nadel.execute(nadelExecutionInput).join()
+        then:
+        def e = thrown CompletionException
+        e.cause instanceof AssertException
+        e.cause.message == "Schema mismatch: Underlying schema is missing required field Mars"
```
Comment (andimarek) [inline src/test/groovy/graphql/nadel/NadelE2ETest.groovy:1045]: this error message doesn't seem completely right: `Mars` is the type not the field.
Maybe we need to differentiate between type and field missing.

Comment (Sajan-99) [reply src/test/groovy/graphql/nadel/NadelE2ETest.groovy:1045]: Yeah sorry this should be Object/Interface

File: src/test/groovy/graphql/nadel/NadelE2ETest.groovy [hunk 2]
```diff
@@ -984,4 +986,66 @@ class NadelE2ETest extends Specification {
...
+                .query(query)
+                .operationName("OpName")
+                .build()
+
+        when:
+        nadel.execute(nadelExecutionInput).join()
+        then:
+        def e = thrown CompletionException
+        e.cause instanceof AssertException
+        e.cause.message == "Schema mismatch: Underlying schema is missing required Interface Type Mars"
```
Comment (bbakerman) [inline src/test/groovy/graphql/nadel/NadelE2ETest.groovy:1045]: I know this ia nit - but the capitalization is inconsistent - why capitalise `Type` say?

How about `Schema mismatch: The underlying schema is missing required interface 'Mars'`

**Pull request merged**